File: 2283-check-if-number-has-equal-digit-count-and-digit-value/2283-check-if-number-has-equal-digit-count-and-digit-value.java
class Solution {
    public boolean digitCount(String num) {
        int[] count = new int[10];
        Arrays.fill(count,0);
        for(int i=0;i<num.length();i++){
            count[num.charAt(i)-'0']++;
        }
        for(int i=0;i<num.length();i++){
            if (num.charAt(i)-'0' != count[i]) return false;
        }
        return true;
    }
}

File: 2283-check-if-number-has-equal-digit-count-and-digit-value/2283-check-if-number-has-equal-digit-count-and-digit-value.js
/**
 * @param {string} num
 * @return {boolean}
 */
var digitCount = function(num) {
        const count = Array(10).fill(0);
        for(let i=0;i<num.length;i++){
            count[num[i]]++;
        }
        for(let i=0;i<num.length;i++){
            if (num[i] != count[i]) return false;
        }
        return true;
};